code/bmpman/bmpman.cpp

```diff
@@ -2576,6 +2609,61 @@ void bm_page_in_start() {
 	gr_bm_page_in_start();
 }
 
+/**
+ * Stuffs Pathtypes with an additional path leading to a subdir of the same name as the given file (minus file extension).
+ * Used for allowing frames of .eff animations to be found in their own subdirectories.
+ *
+ * Example: if called with "debris.eff" and that file is in data/maps, then Pathtypes[CF_TYPE_TEMP_SUBDIR_LOOKUP].path
+ * will be stuffed with data/maps/debris, allowing data/maps/debris/debris_0000.dds (etc) to be found
+ */
+bool set_temp_subdir_pathtype(const char *filename) {
```

Comment:
This was more of a "nice-to-have" request so it's not very important. However, I just noticed that this function is defined in `bmpman.cpp` and not in the CFile system. The functionality of this function should be present in the CFile system and not in bmpman since it changes stuff inside the CFile system.